codex/sales/reservations.nim

```diff
@@ -470,109 +476,116 @@ method createReservation*(
     slotIndex: uint64,
     collateralPerByte: UInt256,
     validUntil: SecondsSince1970,
-): Future[?!Reservation] {.async, base.} =
-  withLock(self.availabilityLock):
-    without availabilityKey =? availabilityId.key, error:
-      return failure(error)
+): Future[?!Reservation] {.async: (raises: [CancelledError]), base.} =
```

Comment:
This made me realise that because createReservation performs multiple async ops, it should not be cancelled when called from the state machine, otherwise the Availability/Reservation may end up in an inconsistent state depending how far along createReservation was at the time of cancellation. So maybe as part of this PR we should also change (in `sales/states/preparing`):
```diff
without reservation =?
-      await reservations.createReservation(
+      await noCancel reservations.createReservation(
        availability.id, request.ask.slotSize, request.id, data.slotIndex,
        request.ask.collateralPerByte, requestEnd,
      ), error:
```